instrumentation/kafka/kafka-connect-2.6/testing/src/test/java/io/opentelemetry/instrumentation/kafkaconnect/v2_6/MongoKafkaConnectSinkTaskTest.java

```diff
@@ -0,0 +1,728 @@
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.shaded.com.google.common.base.VerifyException;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+@Testcontainers
+// Suppressing warnings for test dependencies and deprecated Testcontainers API
+@SuppressWarnings({"rawtypes", "unchecked", "deprecation", "unused"})
+class MongoKafkaConnectSinkTaskTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(MongoKafkaConnectSinkTaskTest.class);
+
+  // Using the same fake backend pattern as smoke tests (with ARM64 support)
+  private static GenericContainer<?> backend;
+
+  private static final String CONFLUENT_VERSION = "7.5.9";
+
+  // Ports
+  private static final int KAFKA_INTERNAL_PORT = 9092;
+  private static final int ZOOKEEPER_INTERNAL_PORT = 2181;
+  private static final int KAFKA_INTERNAL_ADVERTISED_LISTENERS_PORT = 29092;
+  private static final int CONNECT_REST_PORT_INTERNAL = 8083;
+
+  // Network Aliases
+  private static final String KAFKA_NETWORK_ALIAS = "kafka";
+  private static final String ZOOKEEPER_NETWORK_ALIAS = "zookeeper";
+  private static final String KAFKA_CONNECT_NETWORK_ALIAS = "kafka-connect";
+  private static final String MONGO_NETWORK_ALIAS = "mongodb";
+  private static final String BACKEND_ALIAS = "backend";
+  private static final int BACKEND_PORT = 8080;
+
+  // Database
+  private static final String DB_NAME = "testdb";
+  private static final String COLLECTION_NAME = "person";
+
+  // Other constants
+  private static final String PLUGIN_PATH_CONTAINER = "/usr/share/java";
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+  private static final String CONNECTOR_NAME = "test-mongo-connector";
+  private static final String TOPIC_NAME = "test-mongo-topic";
+
+  // Docker network / containers
+  private static Network network;
+  private static FixedHostPortGenericContainer<?> kafka;
```

Comment:
Is using the deprecated `FixedHostPortGenericContainer` necessary? In other tests we use `KafkaContainer`, for example see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/kafka/kafka-clients/kafka-clients-0.11/testing/src/main/java/io/opentelemetry/instrumentation/kafkaclients/common/v0_11/internal/KafkaClientBaseTest.java Perhaps that would help?